src/bluehawk.ts

```diff
@@ -0,0 +1,30 @@
+import { makeBlockCommentTokens } from "./lexer/makeBlockCommentTokens";
+import { makeLineCommentToken } from "./lexer/makeLineCommentToken";
+import { BluehawkError, makeCstVisitor } from "./parser/makeCstVisitor";
+import { validateVisitorResult } from "./parser/validator";
+import { RootParser } from "./parser/RootParser";
+
+export class BluehawkResult {
+  errors: BluehawkError[];
+}
+
+export class Bluehawk {
+  static bluehawk(text: string): BluehawkResult {
```

Comment:
Eh, I don't know about verbifying it more. Bluehawk.bluehawk() seems a bit annoying. Maybe `parse()`?

Also I don't think this should be static or a singleton class. I think the parser and visitor actually should not be created on each run, since that's actually pretty expensive. We'll want to reuse different parsers per language. This is the class where you'll instantiate and store those parsers as needed.

Reply:
Hm, can you point me to any singleton examples in typescript? Most of my experience with singletons has been in the Java world (`@Singleton` ftw) so any pointers would be appreciated!

Ditto on "bluehawk.bluehawk" -- I was debating just using a top-level function instead, but especially if we're dealing with singletons, it sounds like we should change the method name to parse and throw it in a class.

Reply:
It should *not* be a singleton... as it currently only has static methods it may as well be a singleton. Let's avoid that pattern and allow people to create different instances as needed.

Reply:
Fair enough -- done!